pyro/contrib/funsor/infer/tracetmc_elbo.py

```diff
@@ -21,28 +22,25 @@ def _get_trace(self, *args, **kwargs):
         raise ValueError("shouldn't be here")
 
     def differentiable_loss(self, model, guide, *args, **kwargs):
-        with enum(first_available_dim=-self.max_plate_nesting-1):
+        with plate(size=self.num_particles) if self.num_particles > 1 else contextlib2.nullcontext(), \
```

Comment:
nit: Use an empty `ExitStack` as a `nullcontext` to avoid adding a dependency?
```py
try:
    from contextlib import nullcontext
except ImportError:
    from contextlib import ExitStack as nullcontext
```